src/audio/copier/copier.c

```diff
@@ -1749,7 +1705,7 @@ static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *par
 				} else {
 					/* handle gtw case */
 					ret = ipcgtw_zephyr_params(cd->ipcgtw_data[i],
-								   cd->endpoint[i],
```

Comment:
BTW @btian1 if we'd never create a new device for ipc_gtw, do you want to remove all that device related code in ipc_gtw.c as well?

Reply:
I also think about this, like Guennadi did in host, just not sure submit in this pr or a new pr.

Reply:
in this PR as the last patch. Guennadi's patch was blocked because the host dev is needed for IPC3. ipc_gtw is ipc4 specific. so no issues removing it